src/languages/snowflake/snowflake.formatter.ts

```diff
@@ -0,0 +1,352 @@
+import { expandPhrases } from 'src/expandPhrases';
+import { DialectFormatOptions } from 'src/formatter/ExpressionFormatter';
+import Formatter from 'src/formatter/Formatter';
+import Tokenizer from 'src/lexer/Tokenizer';
+import { functions } from './snowflake.functions';
+import { keywords } from './snowflake.keywords';
+
+const reservedSelect = expandPhrases(['SELECT [ALL | DISTINCT]']);
+
+// https://docs.snowflake.com/en/sql-reference/sql-all.html
+//
+// 1. run in console on this page: $x('//tbody/tr/*[1]//a/span/text()').map(x => x.nodeValue)
+// 2. delete all lines that contain a sting like '(.*)', they are already covered in the list
+// 3. delete all lines that contain a sting like '<.*>', they are already covered in the list
+// 4. delete all lines that contain '…', they are part of a regex statement that can't be covered here
+// 5. Manually add 'COPY INTO'
+//
+// Steps 1-4 can be combined by the following script in the developer console:
+// $x('//tbody/tr/*[1]//a/span/text()').map(x => x.nodeValue) // Step 1
+//   filter(x => !x.match(/\(.*\)/) && !x.match(/…/) && !x.match(/<.*>/)) // Step 2-4
+const reservedCommands = expandPhrases([
+  // queries
+  'WITH [RECURSIVE]',
+  'FROM',
+  'WHERE',
+  'GROUP BY',
+  'HAVING',
+  'PARTITION BY',
+  'ORDER BY',
+  'LIMIT',
+  'OFFSET',
+  'FETCH [FIRST | NEXT]',
+  // Data manipulation
+  // - insert:
+  'INSERT [OVERWRITE] [ALL INTO | INTO | ALL | FIRST]',
```

Comment:
Are multiple following `[...]` blocks supported? I'm unsure because of the spaces in between, I dont want it to become something like `INSERT  INTO` (2 spaces in between).

Reply:
Yep that's fine. The spaces will be taken care of by `expandPhrases()`. In the end though any number of whitespace characters between these words will be allowed by the parser.